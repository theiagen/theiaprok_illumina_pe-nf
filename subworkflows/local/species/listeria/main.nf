include { LISSERO  } from '../../../../modules/local/species_typing/listeria/lissero/main'
workflow LISTERIA_SPECIES_TYPING {

    take:
    ch_listeria // Channel of tuples [meta, assembly, reads, species]

    main:

    ch_samples = ch_listeria.map { meta, assembly, reads, species -> [meta, assembly] }

    ch_versions = Channel.empty()
    ch_lissero_results = Channel.empty()

    LISSERO (
        ch_samples,
        params.lissero_min_percent_identity ?: 95.0,
        params.lissero_min_percent_coverage ?: 95.0
    )
    ch_lissero_results = LISSERO.out.lissero_results
    ch_versions = ch_versions.mix(LISSERO.out.versions)


    emit:
    lissero_results = ch_lissero_results

    versions = ch_versions
}