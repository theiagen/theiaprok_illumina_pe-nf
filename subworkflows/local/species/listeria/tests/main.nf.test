nextflow_workflow {
    name "Test Subworkflow LISTERIA_SPECIES"
    script "../main.nf"
    workflow "LISTERIA_SPECIES_TYPING"

    test("listeria_species - listeria characterization") {
        setup {
            run("GUNZIP") {
                script "../../../../../modules/nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'test_listeria', single_end:false],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/848/505/GCF_002848505.1_ASM284850v1/GCF_002848505.1_ASM284850v1_genomic.fna.gz', checkIfExists: true)
                    ])
                    """
                }
            }
        }
        
        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                    .map { meta, assembly -> 
                        [
                            meta, 
                            assembly, 
                            [
                                file("gs://theiagen-large-public-files-rp/terra/theiaprok-files/theiaprok_validation/20012105104_R1.fastq.gz", checkIfExists: true),
                                file("gs://theiagen-large-public-files-rp/terra/theiaprok-files/theiaprok_validation/20012105104_R2.fastq.gz", checkIfExists: true)
                            ],
                            "Listeria"
                        ]
                    }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.lissero_results,
                    workflow.out.versions
                ).match() }
            )
        }
    }
}