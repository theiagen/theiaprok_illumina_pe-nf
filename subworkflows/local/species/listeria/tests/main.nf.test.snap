{
    "listeria_species - listeria characterization": {
        "content": [
            [
                [
                    {
                        "id": "test_listeria",
                        "single_end": false
                    },
                    "test_listeria.tsv:md5,adb40955c2632cd74698b848faa9f59b"
                ]
            ],
            [
                "versions.yml:md5,d38c0c5e9635f599d864eeb8d1e5f1d2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-02T19:51:17.232013654"
    },
    "acinetobacter_species - abaumannii characterization": {
        "content": [
            [
                
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-02T19:47:01.949198489"
    }
}